web/src/composebox_typeahead.ts

```diff
@@ -858,30 +1099,44 @@ export function content_highlighter_html(item) {
         case "stream":
             return typeahead_helper.render_stream(item);
         case "syntax":
-            return typeahead_helper.render_typeahead_item({primary: item});
+            return typeahead_helper.render_typeahead_item({primary: item.language});
         case "topic_jump":
-            return typeahead_helper.render_typeahead_item({primary: item});
+            return typeahead_helper.render_typeahead_item({primary: item.message});
         case "topic_list":
-            return typeahead_helper.render_typeahead_item({primary: item});
+            return typeahead_helper.render_typeahead_item({primary: item.topic});
         case "time_jump":
-            return typeahead_helper.render_typeahead_item({primary: item});
+            return typeahead_helper.render_typeahead_item({primary: item.message});
         default:
             return undefined;
     }
 }
 
-export function content_typeahead_selected(item, query, input_element, event) {
+export function content_typeahead_selected(
+    item: PillItem,
+    query: string,
+    input_element: TypeaheadInputElement,
+    event?: JQuery.ClickEvent | JQuery.KeyUpEvent | JQuery.KeyDownEvent,
+): string {
     const pieces = split_at_cursor(query, input_element.$element);
     let beginning = pieces[0];
     let rest = pieces[1];
-    const $textbox = input_element.$element;
+    assert(input_element.type === "textarea");
+    const $textbox: JQuery<HTMLTextAreaElement> = input_element.$element;
     // Accepting some typeahead selections, like polls, will generate
     // placeholder text that is selected, in order to clarify for the
     // user what a given parameter is for. This object stores the
     // highlight offsets for that purpose.
-    const highlight = {};
-
-    switch (completing) {
+    const highlight: {
+        start?: number;
+        end?: number;
+    } = {};
+    if (completing === "topic_jump") {
+        item = {
```

Comment:
The `item` is still a stream item here, so I replaced it so that the `switch` would handle it properly. I'll look a bit more into why this happens and leave a code comment.